File: fe-admin/src/app/core/model/response/treatment-response.ts
import {TreatmentCategoryResponse} from "./treatment-category-response";

export interface TreatmentResponse{
    id:number;
    code:string;
    name:string;
    description:string;
    price:number;
    unit:string;
    isActive:boolean;
    version:number;
    treatmentCategory:TreatmentCategoryResponse;
    createdAt:Date;
    modifiedAt:Date;
}

File: be-api/src/main/java/cd/beapi/dto/response/TreatmentResponse.java
package cd.beapi.dto.response;

import java.math.BigDecimal;
import java.time.Instant;

public record TreatmentResponse(
        Long id,
        String code,
        String name,
        String description,
        BigDecimal price,
        String unit,
        Boolean isActive,
        Long version,
        TreatmentCategoryResponse treatmentCategory,
        Instant createdAt,
        Instant modifiedAt) {
}
